wled00/e131.cpp

```diff
@@ -162,38 +162,71 @@ void handleE131Packet(e131_packet_t* p, IPAddress clientIP, byte protocol){
         setRealtimePixel(i, e131_data[dataOffset+1], e131_data[dataOffset+2], e131_data[dataOffset+3], wChannel);
       break;
 
-    case DMX_MODE_EFFECT: // Length 1: Apply Preset ID, length 11-13: apply effect config
-      if (uni != e131Universe) return;
-      if (availDMXLen < 11) {
-        if (availDMXLen > 1) return;
-        applyPreset(e131_data[dataOffset+0], CALL_MODE_NOTIFICATION);
-        return;
+    case DMX_MODE_PRESET:       // 2 channel: [Dimmer,Preset]
+      if (uni != e131Universe || availDMXLen < 2) return;
+      applyPreset(e131_data[dataOffset+1], CALL_MODE_NOTIFICATION);
+      if (bri != e131_data[dataOffset]) {
+        bri = e131_data[dataOffset];
+        strip.setBrightness(bri, true);
       }
+      return;
+      break;
 
-      if (bri != e131_data[dataOffset+0]) {
-        bri = e131_data[dataOffset+0];
-      }
-      if (e131_data[dataOffset+1] < strip.getModeCount())
-      effectCurrent   = e131_data[dataOffset+ 1];
-      effectSpeed     = e131_data[dataOffset+ 2];  // flickers
-      effectIntensity = e131_data[dataOffset+ 3];
-      effectPalette   = e131_data[dataOffset+ 4];
-      col[0]          = e131_data[dataOffset+ 5];
-      col[1]          = e131_data[dataOffset+ 6];
-      col[2]          = e131_data[dataOffset+ 7];
-      colSec[0]       = e131_data[dataOffset+ 8];
-      colSec[1]       = e131_data[dataOffset+ 9];
-      colSec[2]       = e131_data[dataOffset+10];
-      if (availDMXLen > 11)
+    case DMX_MODE_EFFECT:           // 11 channels [bri,effectCurrent,effectSpeed,effectIntensity,effectPalette,R,G,B,R2,G2,B2]
+    case DMX_MODE_EFFECT_W:         // 13 channels, same as above but with extra +2 white channels [..,W,W2]
+    case DMX_MODE_EFFECT_SEGMENT:   // 11 channels per segment; max[#] = floor[512/(11+DMXSegmentSpacing)] = 46,42,39, ..
+    case DMX_MODE_EFFECT_SEGMENT_W: // 13 Channels per segment; max[#] = floor[512/(13+DMXSegmentSpacing)] = 39,36,34,32, ..
       {
-        col[3]        = e131_data[dataOffset+11]; //white
-        colSec[3]     = e131_data[dataOffset+12];
+        if (uni != e131Universe) return;
+        bool segmentUpdated = false;
+        bool isSegmentMode = DMXMode == DMX_MODE_EFFECT_SEGMENT || DMXMode == DMX_MODE_EFFECT_SEGMENT_W;
+        uint8_t dmxEffectChannels = (DMXMode == DMX_MODE_EFFECT || DMXMode == DMX_MODE_EFFECT_SEGMENT) ? 11 : 13;
+        for (uint8_t seg = 0; seg < strip.getSegmentsNum(); seg++) {
+          if (isSegmentMode)
+            dataOffset = DMXAddress + seg * (dmxEffectChannels + DMXSegmentSpacing);
+          else
+            dataOffset = DMXAddress;
+          // Modify address for Art-Net data
+          if (protocol == P_ARTNET && dataOffset > 0)
+            dataOffset--;
+          // Skip out of universe addresses
+          if (dataOffset > dmxChannels - dmxEffectChannels)
+            return;
+
+          if (e131_data[dataOffset+1] < strip.getModeCount())
```

Comment:
The approach here is wrong. `col[]` and `colSec[]` are deprecated (and may be removed in the future).
Similar goes for `effectXxxxxx` variables. Their purpose is just informational.

Please use the following snippet:
```c++
Segment &segment = strip.getSegment(id);
segment.setMode(e131_data[...]);
segment.speed = e131_data[...];
...
```